### DISCUSSION TITLE
FATES nutrient integration

### DISCUSSION THREAD

**Discussion by wwieder** [ESCOMP/CTSM#1961] [Ideas]
Given the high priority of integrating FATES nutrient capabilities with CTSM I thought I'd start a discussion thread to scope out this challenge.  I would love for this to include development of an alternative ECA approach for nutrient competition in addition to the RDA approach CTSM already has (see also #1947).  As always, the challenge here is resources, but maybe people like @rgknox @adrifoster @jkshuman & @rosiealice would be interested in starting to discuss how we can start making progress on this? 

Maybe this is also a good place to keep track of all that needs to be done, (although maybe a full project board is more helpful?):

- [ ] Migrate FATES to use the normal BGC call sequence, #1959
- [ ] encapsulate the RD and the new ECA scheme into subroutines 
- [ ] remove `nitrif-denitrif` and `non-nitrif-denitrif` flags, #1356

**Comment by rosiealice** [comment]
- - [ ] - Hi @wwieder. Thanks for this. I guess one thjng I'm not very up to speed with is the status of MIMICS in CLM, how that affects the nutrient competition, and what we are thinking wrt coupling that to FATES. As you know, work on an expanded version of MIMICS is ongoing in Oslo and we are thinking about the most efficient course of action for that to move forwards...

**Reply by wwieder** [reply]
Good question!  MIMICS is up and running in CLM, and the current code is on the main development branch.  We're currently suffering from spin-up challenges that are yet to be resolved.  Preliminary results for single point cases seem reasonable (i.e. the model runs and generates sensible soil C depth profiles).  I honestly haven't had time to look results this with much more detail, it's been a casualty of being overcommitted with LMWG activities and the continued spin-up issues.  

I'd assume creating a branch from CTSM-main that integrates Elin's MIMICS+ work should be pretty straight forward.  That said, maybe it's better to have MIMICS and FATES talking to each other first, which would require this FATES nutrient integration piece to come in first...?  Happy to discuss in more detail.

**Comment by jkshuman** [comment]
linking in the recent work that @rgknox has been doing which acts as a beginning to nutrient coupling work with @adrifoster There is a testing plot included showing the response of integrating this update. This is under discussion and review, so we will discuss further.
https://github.com/ESCOMP/CTSM/pull/1959#issuecomment-1456726988

**Comment by rgknox** [comment]
Two things that might make the process of getting ECA into CLM more viable:
1) encapsulate the RD and the new ECA scheme in subroutines that can be called.  The code where these things happen gets really long and hard to follow. I can port over the work that was done for this in CLM's sister land model
2) having both **nitrif-denitrif** and **non-nitrif-denitrif** also makes the code long, seemingly redundant and easier to get lost. My understanding is that no-one really uses the non-nitrif-denitirif code anymore. It might be nice to also just get rid of it to make the code readable.  We've had threads on this in the past, and it seemed like there was support.

**Reply by wwieder** [reply]
these both seem like sensible suggestions to me, @rgknox.  We'll take a look at the logistics of removing the **non-nitrif-denitrif** options (which I agree are not used any more).

**Reply by wwieder** [reply]
Looks like there are already some threads on the **nitrif-denitrif** issue that were related to previous discussion on how we `use_cn`, #1356.  @olyson, could you look into what would be involved in removing this namelist flag completely from the code?

**Reply by wwieder** [reply]
In the SE meeting today @ekluzek said he'd look into setting `nitrif_denitrif = .true.` so then it can be removed.

**Comment by rgknox** [comment]
@adrifoster and I came up with a list of steps to this process. But it got jumbled when it made more sense to get all of the CLM-side N dynamics code working with FATES in this first pass, albeit with no plant to litter, or soil to plant N exchange.  Here is a rough, 3 step process as I see it:

1) (current) Implement the normal BGC call sequence with FATES active, which includes FATES->CLM C litter flux, C based balance checking, N based balance checking, N fixation, N deposition, but NO FATES->CLM N litter flux or CLM->FATES PLant N uptake
2) Enabled CN-only (no P) FATES to couple with CLM, with actual N litter flux and synthetic/prescribed N uptake (fates side) 
3) Create synthetic or prescribed N uptake from the CLM side?
4) Full N acquisition algorithms
